src/antsibull_build/versions.py

```diff
@@ -333,3 +339,19 @@ def find_latest_compatible(
         )
 
     return reduced_versions
+
+
+def load_all_dependency_files(
+    deps_dir: os.PathLike[str] | str,
+    *,
+    accept_deps_file: t.Callable[[os.PathLike[str] | str, str], bool] | None = None,
+) -> dict[str, DependencyFileData]:
+    dependencies: dict[str, DependencyFileData] = {}
+    for path in glob.glob(os.path.join(deps_dir, "*.deps"), recursive=False):
```

Comment:
Nit: I would use `iglob` (returns an iterator of paths instead of a list) here to avoid creating an unnecessary list (`glob` basically just calls `list(iglob(*args, **kwargs))`). Also, `recursive=False` is already the default and only applies when using `**` in a pattern which this code does not.

```suggestion
    for path in glob.iglob(os.path.join(deps_dir, "*.deps")):
```